addons/binding/org.openhab.binding.omnikinverter/src/main/java/org/openhab/binding/omnikinverter/handler/OmnikInverterHandler.java

```diff
@@ -0,0 +1,103 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.omnikinverter.OmnikInverter;
+import org.openhab.binding.omnikinverter.OmnikInverterBindingConstants;
+import org.openhab.binding.omnikinverter.OmnikInverterConfiguration;
+import org.openhab.binding.omnikinverter.OmnikInverterMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OmnikInverterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans van den Bogert - Initial contribution
+ */
+@NonNullByDefault
+public class OmnikInverterHandler extends BaseThingHandler {
+    @Nullable
+    private OmnikInverter inverter;
+
+    private final Logger logger = LoggerFactory.getLogger(OmnikInverterHandler.class);
+
+    public OmnikInverterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(OmnikInverterBindingConstants.CHANNEL_POWER)) {
+            if (command instanceof RefreshType) {
+                updateData();
+            }
+        }
+    }
+
+    private void updateData() {
+        try {
+            if (inverter != null) {
+                OmnikInverterMessage message = inverter.pullCurrentStats();
+
+                updateStatus(ThingStatus.ONLINE);
+
+                QuantityType<Power> powerQuantity = new QuantityType<>(message.getPower(), SmartHomeUnits.WATT);
+                updateState(OmnikInverterBindingConstants.CHANNEL_POWER, powerQuantity);
+
+                updateState(OmnikInverterBindingConstants.CHANNEL_ENERGY_TODAY,
+                        new QuantityType<Energy>(message.getEnergyToday(), SmartHomeUnits.KILOWATT_HOUR));
+            } else {
+                return;
+            }
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unknown host provided");
+        } catch (java.net.NoRouteToHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "No route to host");
+        } catch (ConnectException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Host does not allow socket connection");
+        } catch (IOException e) {
+            logger.debug(e.getMessage());
```

Comment:
Logger statements have possibility to specify a message - generally it is good practice to add your own message when logging error even if you got it from exception. Sometimes exception messages are bit cryptic, sometimes they are even empty. 
```suggestion
            logger.debug("Could not connect and/or fetch data from inverter", e);
```